src/gui/sdlmain.cpp

```diff
@@ -2229,6 +2241,71 @@ static SDL_Point window_bounds_from_label(const std::string &pref,
 	return {w, h};
 }
 
+static SDL_Point clamp_to_minimum_window_dimensions(SDL_Point size)
+{
+	const auto w = std::max(size.x, FALLBACK_WINDOW_DIMENSIONS.x);
+	const auto h = std::max(size.y, FALLBACK_WINDOW_DIMENSIONS.y);
+	return {w, h};
+}
+
+static SDL_Rect get_desktop_resolution()
+{
+	SDL_Rect desktop;
+	assert(sdl.display_number >= 0);
+	SDL_GetDisplayBounds(sdl.display_number, &desktop);
+	assert(desktop.w >= FALLBACK_WINDOW_DIMENSIONS.x);
+	assert(desktop.h >= FALLBACK_WINDOW_DIMENSIONS.y);
+	return desktop;
+}
+
+// Takes in:
+//  - The 'maxresolution' config value: 'default', 'W,H', or an invalid setting.
+//
+// Except for SURFACE and TEXTURE rendering, the function populates the following struct members:
+//  - 'sdl.desktop.restrict_resolution', true if resolution restriction is enabled.
+//  - 'sdl.desktop.max_resolution', with the refined size.
+
+static void setup_resolution_restriction_from_conf(const char *max_resolution_val)
+{
+	// TODO: Deprecate SURFACE output and remove this.
+	if (sdl.desktop.want_type == SCREEN_SURFACE)
+		return;
+
+	// TODO: Implement support for TEXTURE output.
+	if (sdl.desktop.want_type == SCREEN_TEXTURE)
+		return;
+
+	const auto desktop = get_desktop_resolution();
+	int w, h;
+	auto was_parsed = sscanf(max_resolution_val, "%dx%d", &w, &h) == 2;
+
+	sdl.restrict_resolution = false;
+
+	const bool is_out_of_bounds = (w > desktop.w || h > desktop.h);
+	if (was_parsed && is_out_of_bounds) {
+		LOG_MSG("DISPLAY: Requested maxresolution '%dx%d' is larger than the desktop '%dx%d'",
+		        w, h, desktop.w, desktop.h);
+		return;
+	}
+
+	if (was_parsed && w > 0 && h > 0) {
```

Comment:
Negative width or height also implies being out-of-bounds, so I would collapse this failure criteria with the prior "out-of-bounds" state.

```c++
const bool is_out_of_bounds = w <= 0 || w > desktop.w || h <= 0 || h > desktop.h;
```

Maybe mention something like, `"is outside the bounds of the desktop' '%dx%d'"`.

Reply:
👍🏻